src/client/src/Stream.c

```diff
@@ -1389,7 +1404,7 @@ STATUS getStreamData(PKinesisVideoStream pKinesisVideoStream, UPLOAD_HANDLE uplo
             // Map the storage
             CHK_STATUS(heapMap(pKinesisVideoClient->pHeap, pKinesisVideoStream->curViewItem.viewItem.handle, (PVOID *) &pAlloc, &allocSize));
             CHK(allocSize < MAX_UINT32, STATUS_INVALID_ALLOCATION_SIZE);
-            size = (UINT32) allocSize;
+            size = pKinesisVideoStream->curViewItem.viewItem.length;
 
             // Validate we had allocated enough storage just in case
             CHK(pKinesisVideoStream->curViewItem.viewItem.length - pKinesisVideoStream->curViewItem.offset <= size, STATUS_VIEW_ITEM_SIZE_GREATER_THAN_ALLOCATION);
```

Comment:
Not sure I follow. The allocSize is a 64 bit integer whereas size is 32 so we add a check to validate that our frames are less than 32 bit in the first place.

Then, we actually use the length in the view item. Perhaps there is a diff tool issue but here is the code in this area:



CHK_STATUS(heapMap(pKinesisVideoClient->pHeap, pKinesisVideoStream->curViewItem.viewItem.handle, (PVOID *) &pAlloc, &allocSize));
            CHK(allocSize < MAX_UINT32, STATUS_INVALID_ALLOCATION_SIZE);
            size = pKinesisVideoStream->curViewItem.viewItem.length;

Reply:
Oh, I was referring to `CHK(pKinesisVideoStream->curViewItem.viewItem.length - pKinesisVideoStream->curViewItem.offset <= size, STATUS_VIEW_ITEM_SIZE_GREATER_THAN_ALLOCATION);`

Since size is `pKinesisVideoStream->curViewItem.viewItem.length`, the check is only validating,

pKinesisVideoStream->curViewItem.viewItem.length - pKinesisVideoStream->curViewItem.offset <= **pKinesisVideoStream->curViewItem.viewItem.length**

which is equal to `pKinesisVideoStream->curViewItem.offset >= 0`, which should be always true since `curViewItem.offset` is unsigned integer. (except when pKinesisVideoStream->curViewItem.viewItem.length - pKinesisVideoStream->curViewItem.offset wraps around).

Reply:
Yes indeed, in modified code it makes lesser sense. However, this check was always an internal error catching check. In this case, it would sanitize the value of the offset. I would still leave it there though, as you say, it doesn't really add much value right now but could be a deterrent against bad code in the future that could cause weird behavior. 

We are extra careful when it comes to content store allocations. In the normal run time release heap, the allocations will be tightly packed and the overruns will not generate any crashes on the client but will corrupt the data in a weird way.

Reply:
Right, I was not suggesting to delete the check, but rather updating it to be 

`CHK(pKinesisVideoStream->curViewItem.viewItem.length - pKinesisVideoStream->curViewItem.offset <= allocSize, STATUS_VIEW_ITEM_SIZE_GREATER_THAN_ALLOCATION);`

This check will guarantee that we have enough memory and will not overrun.

Also, `size = pKinesisVideoStream->curViewItem.viewItem.length;` should be removed since it'll be overwritten in the line after by

https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/678c208718985789e6db5f18f9437b5380290026/src/client/src/Stream.c#L1413

I probably missed something, feel free to ignore it 😊

Reply:
Oh, right. Uploading the change now